core/src/main/java/jenkins/util/SystemProperties.java

```diff
@@ -0,0 +1,253 @@
+      */  
+    public static boolean getBoolean(String name) {
+        return getBoolean(name, false);
+    }
+
+    /**
+      * Returns {@code true} if the system property
+      * named by the argument exists and is equal to the string
+      * {@code "true"}, or a default value. If the system property does not exist, return
+      * {@code "true"} if a property by this name exists in the {@link ServletContext}
+      * and is equal to the string {@code "true"}. If that property does not
+      * exist either, return the default value.
+      * 
+      * This behaves just like {@link Boolean#getBoolean(java.lang.String)} with a default
+      * value, except that it also consults the {@link ServletContext}'s "init" parameters.
+      * 
+      * @param   name   the system property name.
+      * @param   def   a default value.
+      * @return  the {@code boolean} value of the system property.
+      */
+    public static boolean getBoolean(String name, boolean def) {
+        String v = getString(name);
+       
+        if (v != null) {
+            return Boolean.parseBoolean(v);
+        }
+        return def;
+    }
+    
+    /**
+      * Determines the integer value of the system property with the
+      * specified name.
+      * 
+      * This behaves just like {@link Integer#getInteger(java.lang.String)}, except that it
+      * also consults the {@link ServletContext}'s "init" parameters.
+      * 
+      * @param   name property name.
+      * @return  the {@code Integer} value of the property.
+      */
+    @CheckForNull
+    public static Integer getInteger(String name) {
+        return getInteger(name, null);
+    }
+
+    /**
+      * Determines the integer value of the system property with the
+      * specified name, or a default value.
+      * 
+      * This behaves just like <code>Integer.getInteger(String,Integer)</code>, except that it
+      * also consults the <code>ServletContext</code>'s "init" parameters. If neither exist,
+      * return the default value.
+      * 
+      * @param   name property name.
+      * @param   def   a default value.
+      * @return  the {@code Integer} value of the property.
+      */
+    public static Integer getInteger(String name, Integer def) {
+        String v = getString(name);
+       
+        if (v != null) {
+            try {
+                return Integer.decode(v);
+            } catch (NumberFormatException e) {
+                // Ignore, fallback to default
+                LOGGER.log(Level.CONFIG, "Property. Value is not integer: {0} => {1}", new Object[] {name, v});
+            }
+        }
+        return def;
+    }
+
+    /**
+     * Invoked by WebAppMain, tells us where to get the "init" parameters from.
+     * 
+     * @param context the <code>ServletContext</code> obtained from <code>contextInitialized</code>
+     */
+    public static void initialize(ServletContext context) {
+        theContext = context;
+    }
+    
```

Comment:
Any class that it's initialized before `WebAppMain` making this call that class `static final` values from `SystemProperties` is not going to look in the context, which is fair enough, but maybe could be documented somewhere to avoid "surprises"